File: 14-longestCommonPrefix.js
console.log('start');

var longestCommonPrefix = function(strs) {
    let prefix = '';
    let currentLetter ='';
    for(i=0;i<strs[0].length;i++){
        currentLetter = strs[0][i];
        for (j=0; j<strs.length;j++){
            if(strs[j][i]!=currentLetter){
                return prefix;
            }
        }
        prefix += currentLetter;
    }
    return prefix;
};

let tests = [];
let solutions = [];

tests.push(['florida','flatter','fluorish']);
solutions.push('fl');

let result = '';
for (let i=0; i<tests.length; i++){
    result = longestCommonPrefix(tests[i]);
    console.log('Test ' + i,result,solutions[i],result==solutions[i]);
}

File: python/14_Longest_Common_Prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if len(strs[0]) < 1: return ''

        ans, letter = '', ''
        match = True
        counter = 0
        while match:
            ans += letter
            if len(strs[0]) <= counter: break
            letter = strs[0][counter]
            for el in strs:
                if len(el)<= counter:
                    match = False
                    break
                if el[counter] != letter:
                    match = False
                    break
            counter += 1

        return ans

tests = []
solutions = []

tests.append(['sdljk','sdf','sdr'])
solutions.append('sd')

tests.append(['sdljk','sdf',''])
solutions.append('sd')

tests.append([])
solutions.append('')

for i in range(len(tests)):
    result = Solution().longestCommonPrefix(tests[i])
    print('Test ' + str(i), 'Result', result, solutions[i])